### ISSUE TITLE
Waku v2 - Nangang internal testnet

### ISSUE THREAD

**Issue by oskarth** [vacp2p/research#43]
This is the first internal testnet for Waku v2.

## Goals

1. Having Waku run for "real" to expose issues early
2. Act as a coordination point for developers, milestone with clear scope and delivery
3. Pave the way for future testnets with more features and participants

## Scope

Scope is as minimal as we can get away with. This is deliberate, in order to get it running as soon as possible.

Not in scope:
- Client integration
- Encryption of messages
- Bridging to v1
- Multiple clients
- Metrics, etc
- Accounting, etc (track 3)
- Efficient routing (track 2)
- Probably no store/filter protocol (though please prove more wrong!)

## Where does the name come from?

Nangang (南港) Station is a metro station in Taipei. https://en.wikipedia.org/wiki/Nangang_station

## When will it start?

Plan is for this week, probably Wednesday or Thursday. This assumes no major blockers are found.

## How long will it run for?

As soon as a quantum of utility exist for other features/infrastructure, so probably just a week or so.

## Tasks

1) Ensure two nodes can connect to each other directly and use publish/subscribe to talk
-- connecting works with quicksim so should be fine
-- e.g. with examples/quicksim style RPC
2) Ensure two nodes can discover each other via third node
3) Deploy node to public place (cluster or ad hoc)
--- make sure this is known location that can be hardcoded (bootstrap/static node)
4) Sanity check for developers to talk over a single topic
6) Update spec to draft mode
7) Mini update people about this (Vac forum/tweet)
-- no big thing, this is embarrassingly limited in scope (on purpose)

...fill in more in comments/edit issue directly.

## Acceptance criteria

A testnet of Waku v2 is running that:

1. Primary Waku developers are connected to (Dean, Kim, Oskar)
2. Allows for communication of `WakuMessage`s over `relay` protocol for a specific topic
-- i.e. publish and subscribe with basic RPC
-- incoming messages can simply be logged to STDOUT
3. Some node running in a publicly connectable location
-- bootstrap node
-- relay node

If we can get basic `store` and `filter` support that'd be great, but it isn't going to block the launch.

4. Spec in draft mode
-- at least for main and relay, may mean leaving store and filter in raw mode if they are not operational

**Comment by oskarth** [comment]
cc @kdeme @decanus

**Comment by oskarth** [comment]
Brief update: basic scripts work.

Some blockers:
https://github.com/status-im/nim-waku/issues/147
https://github.com/status-im/nim-waku/issues/148

Possibly https://github.com/status-im/nim-waku/issues/143

**Comment by oskarth** [comment]
@decanus @kdeme I'd appreciate you helping out with getting this testnet out and working.

**Comment by oskarth** [comment]
Issues before addressed, main blocker is this: https://github.com/status-im/nim-waku/issues/155

Spec issue slightly separate but there's an issue here: https://github.com/vacp2p/specs/issues/178 I'll look at it tomorrow.

**Comment by oskarth** [comment]
Above blocker resolved.

Blocked by @decanus and @kdeme also running it so we can test end to end.

Spec draft PR here https://github.com/vacp2p/specs/issues/178 but once above is addressed we can close this I believe.

**Comment by kdeme** [comment]
Tested and works.

**Comment by oskarth** [comment]
Specs merged too, closing as done.
